python/unityagents/environment.py

```diff
@@ -465,6 +465,11 @@ def _get_state(self, output: UnityRLOutput) -> (AllBrainInfo, bool):
             else:
                 [x.memories.extend([0] * (memory_size - len(x.memories))) for x in agent_info_list]
                 memory = np.array([x.memories for x in agent_info_list])
+            mask_actions = np.ones((len(agent_info_list), sum(self.brains[b].vector_action_space_size)))
+            for agent_index, agent_info in enumerate(agent_info_list):
+                if agent_info.action_mask is not None:
+                    for k in range(len(agent_info.action_mask)):
+                        mask_actions[agent_index, k] = 0 if agent_info.action_mask[k] else 1
```

Comment:
Ok, this is why it is confusing. In C# the mask is a boolean array with false if no mask and true if mask.
On python masks (in the buffer as well) are floats. Since we multiply this mask with whatever 1 means __NOT__ masked.

Reply:
But one the `else 1` note, you are right, it is not needed since we already have an array on ones

Reply:
@vincentpierre - mask is intended to have a 1 if we want to make, so the c# term is right. For python, maybe use inverted_mask